readme-sync/v0/senseml-reference/250 - preprocessors/105 - merge-lines.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "Merge lines"
+hidden: false
+---
+
+
+Merges nearby lines more aggressively than the built-in line merger. Use this preprocessor when the PDF contains oversplit lines, lines overlapping on the x-axis, or "jittery" lines misaligned on the y-axis.
+
+Parameters
+----
+
+| key                                        | value                                  | description                                                  |
+| ------------------------------------------ | -------------------------------------- | ------------------------------------------------------------ |
+| `type` (**required**)                      | string                                 | "mergeLines"                                                 |
+| `directlyAdjacentThreshold` (**required**) | number >= 0.16  default: 0.16          | Fraction of line height under which two adjacent lines (i.e., horizontally distributed along an x-axis) are merged without a space. For example, at 0.15 two lines are merged without a space in between their contents when they are less than 15% of the line height apart from one another.<br/>Under the hood, Sensible uses the default setting for this parameter to transform separate tokens output from OCR into lines. It's relatively uncommon to need to adjust the default value for this parameter. |
```

Comment:
Technically there's no default of 0.16 since it's a required field. I think a possibly better way to talk about the built in line merger is to say that the line merge preprocessor always occurs after a standard line merge with 0.16, 0.6, 1, and positive infinity as the params, so certain line merges will be a no-op (like asking for a directlyAdjacentThreshold below 0.6 when yOverlapThreshold is 1, since any lines 0.6 or less will have already been merged with a space)